### PR TITLE
feat: Add coordinate_system parameter to compute_tangential_and_cross_components and _compute_lensing_angles functions

### PR DESCRIPTION
**Pull request by caioolivv**
Add option to choose coordinate system of ellipticity components.

### CODE REVIEW

**Comment by caioolivv** [comment]
@marina-ricci hey, I think the PR is ready for a review!

File: clmm/dataops/__init__.py [hunk 1]
```diff
@@ -376,12 +389,16 @@ def _compute_lensing_angles_flatsky(ra_lens, dec_lens, ra_source_list, dec_sourc
         phi[angsep == 0.0] = 0.0
     else:
         phi = 0.0 if angsep == 0.0 else phi
+    if coordinate_system == "sky":
+        phi = np.pi - phi
```
Comment (hsinfan1996) [inline clmm/dataops/__init__.py:393]: `phi = np.arctan2(deltay, deltax)` should do the work. (note: for coordinate_system == "pixel", the calculation is `phi = np.arctan2(deltay, -deltax)`)

**Comment by caioolivv** [comment]
Since @hsinfan1996 mentioned that it was better to keep things consistent between _compute_lensing_angles_flatsky and _compute_lensing_angles_astropy, it may be better to revert the last commit.

**Comment by hsinfan1996** [comment]
> Since @hsinfan1996 mentioned that it was better to keep things consistent between _compute_lensing_angles_flatsky and _compute_lensing_angles_astropy, it may be better to revert the last commit.

Yeah, I am sorry. I was thinking that the calculation would be easier by flipping the sign of `deltax` for the falt-sky version. However, we only use the astropy version right now, so it may be better to keep the calculations consistent. i will let other reviewers to decide which practice is better.

**Comment by marina-ricci** [comment]
I asked @caioolivv to change the notebooks and add an example in the mock_data notebook for the two definitions. @caioolivv let me know when the PR is ready for review !

**Comment by caioolivv** [comment]
Everything's ready @marina-ricci !

File: clmm/galaxycluster.py [hunk 2]
```diff
@@ -35,6 +37,9 @@ class GalaxyCluster:
         Redshift of galaxy cluster center
     galcat : GCData
         Table of background galaxy data containing at least galaxy_id, ra, dec, e1, e2, z
+    coordinate_system : str
```
Comment (marina-ricci) [inline clmm/galaxycluster.py:40]: You have to specify which one is the default value.

File: clmm/support/mock_data.py [hunk 3]
```diff
@@ -156,6 +157,9 @@ def generate_galaxy_catalog(
         The number density of galaxies (in galaxies per square arcminute, from z=0 to z=infty).
         The number of galaxies to be drawn will then depend on the redshift distribution and
         user-defined redshift range.  If specified, the ngals argument will be ignored.
+    coordinate_system : str, optional
```
Comment (marina-ricci) [inline clmm/support/mock_data.py:160]: same here, specify the default  value

Comment (marina-ricci) [reply clmm/support/mock_data.py:160]: Also, could you add a reference for these definitions? Like a paper?

**Review by marina-ricci — CHANGES_REQUESTED** [review]
Thank you @caioolivv for this important PR! Since the coordinate system is an important but often missed subtlety, we should clearly highlight it. Hence I suggest : 
- to add a references ( like papers) for the definitions in the code (in comments) and in the notebooks ;
- to highlights the differences between the cordinate systyems and profiles in `demo_dataops_functionality.ipynb`
- to expand also the `demo_mock_cluster.ipynb` notebook.

**Comment by caioolivv** [comment]
Ready for review @marina-ricci

**Comment by marina-ricci** [comment]
@caioolivv I think there is  a typo in `demo_dataops_functionality` notebook that is breaking the code. "coordinate system"=piru

**Comment by caioolivv** [comment]
Sorry @marina-ricci , was testing the paramenter validation and forgot to go back to the original value.

**Comment by marina-ricci** [comment]
Ok, everything seems good now !

**Review by marina-ricci — APPROVED** [review]
It lokks good thank you Caio!!!!

**Pull request merged**